ThePalaceProject/circulation
# Add s3 analytics provider

## Description
This PR adds an S3 Analytics Provider which enables users to capture circulation events as json files in S3.  One event, one file. 
This PR is based on https://github.com/ThePalaceProject/circulation/pull/13.  I resolved the conflicts and made a bunch of small tweaks and additions to get it working.  

## Motivation and Context

https://www.notion.so/lyrasis/Sync-circulation-data-to-Redshift-5d28927cce67451caadc9eb7c8eb0936



## How Has This Been Tested?

There are a bunch of tests included with the PR.   To test manually you must:
1. Create a bucket in S3 for analytics.
2. Configure an S3 Storage provider with the above analytics bucket specified.
3. Configure an S3 Analytics Provider
4. Associated S3 Storage Provider with Analytics Provider.
5. Perform a circulation action such as borrowing a book.
6. Confirm that the event is recorded as a Json file in the S3 bucket.
## Checklist
